PhysicsTools/PatAlgos/plugins/PATMuonProducer.cc

```diff
@@ -684,15 +685,15 @@ void PATMuonProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
           << "No primary vertex available from EventSetup, not adding high level selection \n";
     }
     // this is needed by the IPTools methods from the tracking group
-    iSetup.get<TransientTrackRecord>().get("TransientTrackBuilder", trackBuilder);
+    trackBuilder = &iSetup.getData(transientTrackBuilderToken_);
   }
 
   // MC info
   edm::Handle<edm::ValueMap<reco::MuonSimInfo>> simInfo;
   bool simInfoIsAvailalbe = iEvent.getByToken(simInfo_, simInfo);
 
   // this will be the new object collection
-  std::vector<Muon>* patMuons = new std::vector<Muon>();
+  auto patMuons = std::make_unique<std::vector<Muon>>();
```

Comment:
For future reference, this could have just been `std::vector<Muon> pathMuons` and then call `iEvent.emplace(std::move(pathMuons))`.